src/components/Tutorial/TutorialNavigation.tsx

```diff
@@ -51,10 +70,7 @@ const TutorialNavigation = ({
             const step = i + (tutorialStep < TUTORIAL2_STEP_START ? 1 : TUTORIAL2_STEP_START)
             return (
               <a
-                className={classNames({
-                  'tutorial-step-bullet': true,
-                  active: step === Math.floor(tutorialStep),
-                })}
+                className={tutorialStepBullet({ active: step === Math.floor(tutorialStep) })}
```

Comment:
I can think of a couple ways to do this with inline styles, but correct me if I'm wrong:

```ts
css({
    color: 'inherit',
    textDecoration: 'none',
    fontSize: '32px',
    opacity: step === Math.floor(tutorialStep) ? 0.25 : undefined,
    marginLeft: '1px',
    marginRight: '1px',
    transition: 'all 400ms ease-in-out',
  })
```

or

```ts
css({
    color: 'inherit',
    textDecoration: 'none',
    fontSize: '32px',
    marginLeft: '1px',
    marginRight: '1px',
    transition: 'all 400ms ease-in-out',
  }, {
opacity: step === Math.floor(tutorialStep) ? 0.25 : undefined
})
```

The advantage in my mind is that the styles are closer to where they are being used, it depends less on panda-specific syntax, it avoids a one-time use identifier. Let me know if there's something I'm missing though! I'm still learning about PandaCSS and finding the right balance. We ultimately want to take advantage of the Panda compiler without adopting too many Panda-specific idioms unless they provide a major benefit.

Reply:
I've fixed this now. Yes, doing this makes sense. I've just run into so many cases where panda wasn't statically generating classes with conditional statements that I started automatically turning to variants instead. I did this this though, and confirmed it works. I guess panda can handle the simpler conditional statements.

Reply:
Great, thank!

Yeah, I'm still unsure all the cases where static analysis is possible. It's definitely a different way of thinking about the code. PandaCSS can handle simple ternary statements at least when the values are static. It may be worth looking into the generated CSS to see how it compares, but that’s more of an optimization.

I would say that variants are good for well-defined variations on a recipe that is used in multiple modules, whereas inline css is better for ad hoc styles. Others who value standardization over co-colocation may make different architectural decisions though.